core/htmlhelper.py

```diff
@@ -411,6 +411,48 @@ def style_links(soup, class_name):
     return soup
 
 
+def add_class_to_sibling_by_header(
+    soup, header_text, class_to_add, header_tag="h3", target_tag="ul"
+):
+    """
+    Adds a class to the next sibling of a specified tag
+    if it matches the search criteria.
+    """
+    sections = soup.find_all(header_tag, string=header_text)
+    for section in sections:
+        target = section.find_next_sibling(target_tag)
+        if target:
+            existing_classes = target.get("class", [])
+            if class_to_add not in existing_classes:
+                target["class"] = existing_classes + [class_to_add]
+                return target
+    return None
+
+
+def reformat_new_libraries_list(soup):
+    new_libraries_list = add_class_to_sibling_by_header(
+        soup, "New Libraries", "new-libraries", search_tag="h3", target_tag="ul"
```

Comment:
just to make it clearer for the future, it'd be better to pass the argument names here (e.g. `class_to_add="new-libraries" ...)`

Reply:
Gotcha, makes sense! Thanks

Reply:
also, `return None` happens automatically if nothing else is returned :)

Reply:
TY, Lord Kenneth. Removed statement in the latest commit